lib_nn/src/cpp/filt2d/geom/ImageGeometry.cpp

```diff
@@ -0,0 +1,49 @@
+#include "geom/ImageGeometry.hpp"
+
+using namespace nn;
+
+int ImageGeometry::Index(const int row, const int col,
+                         const int channel) const {
+  if (row < 0 || row >= this->height) return -1;
+  if (col < 0 || col >= this->width) return -1;
+  if (channel < 0 || channel >= this->depth) return -1;
+
+  return this->depth * (this->width * row + col) + channel;
```

Comment:
Similarly, is `this->height` missing here?

Reply:
as above